code/modules/power/supermatter/supermatter.dm

```diff
@@ -1025,6 +1038,26 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 	COOLDOWN_START(src, common_radio_cooldown, SUPERMATTER_COMMON_RADIO_DELAY)
 	return TRUE
 
+/obj/machinery/power/supermatter_crystal/proc/holiday_lights()
+	holiday_lights = TRUE
+	RegisterSignal(src, COMSIG_ATOM_ATTACKBY, PROC_REF(holiday_attackby))
```

Comment:
Suggestion: Try using the new `COMSIG_ATOM_ITEM_INTERACTION` rather than attack procs.

Reply:
COMSIG_ATOM_ITEM_INTERACTION doesn't work. Is it because the santa hat isn't a tool?

Reply:
`COMSIG_ATOM_ITEM_INTERACTION` is sent regardless of something's a tool or not, so that's unusual

Reply:
Thanks, yeah fixed the args on it and it works.